### DISCUSSION TITLE
Try to config no multiline f/t when using macro

### DISCUSSION THREAD

**Discussion by jackysee** [ggandor/lightspeed.nvim#84] [Q&A] [answered]
I tried to follow https://github.com/ggandor/lightspeed.nvim#notes to configure the plugin to use native ft during macro staff. It's just copy and paste. But both native ft and lightspeed ft don't work. Have I missed anything?
```vim
" test.vim
if has('vim_starting')
  set encoding=utf-8
endif
scriptencoding utf-8

if &compatible
  set nocompatible
endif

let s:plug_dir = expand('/tmp/plugged/vim-plug')
if !filereadable(s:plug_dir .. '/plug.vim')
  execute printf('!curl -fLo %s/autoload/plug.vim --create-dirs https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim', s:plug_dir)
end

execute 'set runtimepath+=' . s:plug_dir
call plug#begin(s:plug_dir)
Plug 'ggandor/lightspeed.nvim'
call plug#end()
PlugInstall | quit

nmap <expr> f reg_recording() . reg_executing() == "" ? "<Plug>Lightspeed_f" : "f"
nmap <expr> F reg_recording() . reg_executing() == "" ? "<Plug>Lightspeed_F" : "F"
nmap <expr> t reg_recording() . reg_executing() == "" ? "<Plug>Lightspeed_t" : "t"
nmap <expr> T reg_recording() . reg_executing() == "" ? "<Plug>Lightspeed_T" : "T"
```

```nvim -u test.vim test.vim```

**Comment by ggandor** [comment]
The problem might be that you `quit` after `PlugInstall`? 😉 (Although Lightspeed's default mappings should work anyway.)

**Reply by jackysee** [reply]
I tried to comment out the `PlugInstall` line and  manually run `:PlugInstall`, quit and re-opened nvim. It's the same. 
The behavior is that press e.g. `fe` highlight letter `e` but press `f` again won't jump to next occurence.

**Reply by ggandor** [reply]
Btw., there's no need to run `PlugInstall` in the config script every time, it's enough to do it once, manually, whenever you install a new plugin. Also, quitting & re-opening is only needed for remote plugins (e.g. with Python sources).

**Reply by jackysee** [reply]
It's a minimal script template I used to report problem.

**Comment by ggandor** [comment, marked as answer]
I tried it, and you're right, actually! This workaround screws up same-key repeat, I haven't thought about that, and for some reason no one has complained yet :) So for the moment, it only works if you set custom repeat keys in `opts`, but it could only be one key per direction, obviously - i.e., `f`/`F` will repeat all 1-char searches, including `t`/`T`.  (Or you can use `;`/`,` of course.)

`require'lightspeed'.setup({instant_repeat_fwd_key = 'f', instant_repeat_bwd_key = 'F'})`

Thanks for reporting this! I hope we'll come up with a permanent solution to the macro issue soon.

**Reply by jackysee** [reply]
I found that if I want to use native f by `silent! unmap f`. I also need to do the same setup.

```
require'lightspeed'.setup({instant_repeat_fwd_key = 'f', instant_repeat_bwd_key = 'F'})
silent! unmap f
```
